docroot/modules/custom/uiowa_core/uiowa_core.module

```diff
@@ -134,6 +135,131 @@ function template_preprocess_uiowa_core_headline(&$variables) {
+
+  // Separate the attributes of the tag string in to an array of attribute strings.
+  // Ex: ['foo="bar"', 'ban="baz"'].
+  $drupal_media_attributes_strings = explode(' ', $drupal_media_attributes_string);
+
+  // Create a new array to be used later for the associative media attributes data.
+  $associative_drupal_media_attributes_strings = [];
+
+  // For each of the media attributes strings...
+  foreach ($drupal_media_attributes_strings as $value) {
+
+    // Separate them in to two parts, clean them up, and add them to the associative array.
+    $attribute__value = explode('=', $value);
+    $associative_drupal_media_attributes_strings[$attribute__value[0]] = str_replace('"', '', $attribute__value[1]);
+  }
+
+  $view_mode_class = 'inline--size-default';
+
+  // If we have a view mode set...
+  if (isset($associative_drupal_media_attributes_strings['data-view-mode'])) {
+    // Process the view mode classes.
+    $view_mode_modified = uiowa_core_inline_viewmode_map($associative_drupal_media_attributes_strings['data-view-mode']);
+
+    if (!empty($view_mode_modified)) {
+
+      // Replace the $view_mode_class with the relevant modified inline view mode class.
+      $view_mode_class = $view_mode_modified['class'];
+    }
+  }
+  // Get the current classes of the '<drupal-media>' element.
+  $classes_array = explode(' ', $variables['classes']);
+
+  // Add the view mode to the classes.
+  $classes_array[] = $view_mode_class;
+
+  // Re-concatenate the $classes_array back in to a string,
+  // which is what the '<drupal-media>' element wants.
+  $classes = implode(' ', $classes_array);
+
+  // Attach the inline sizing library and assign the classes.
+  $variables['#attached']['library'][] = 'uids_base/inline';
+  $variables['classes'] = $classes;
+
+  // Then, add a custom attribute to the '<drupal-media>' markup to show that it is a figure.
+  $variables['node'] = FilteredMarkup::create(preg_replace(
+    '/<drupal-media /i',
+    '<drupal-media data-figure="true" ',
+    $variables['node']));
+}
+
+/**
+ * Provides mapping for a $view_mode to an inline view mode.
+ *
+ * @param string $view_mode
+ *   A string that is going to be checked
+ *   as a view mode, often looks something
+ *   like 'medium__no_crop'.
+ *
+ * @return array
+ *   If the $view_mode starts with any one of the $accepted_view_modes:
+ *    Returns an array containing the string reference to
+ *    the inline library and the inline sizing class.
+ *   Otherwise, returns an empty array.
+ */
+function uiowa_core_inline_viewmode_map(string $view_mode): array {
+  $accepted_view_modes = ['default', 'large', 'medium', 'small'];
+
+  foreach ($accepted_view_modes as $value) {
+    if (str_starts_with($view_mode, $value)) {
+
+      // We return a string reference to the inline library and a modified view mode class.
+      return [
+        'library' => 'uids_base/inline',
+        'class' => 'inline--size-' . $value,
+      ];
+    }
+  }
```

Comment:
This (`^`) checks for something at the start of the search string, which we can check for each of the values in one go and if found, passes it on for the return.

```suggestion
if (preg_match('%^(' . implode ('|', $accepted_view_modes) . ')%i', $view_mode, $value)) {
  return [
    'library' => 'uids_base/inline',
    'class' => 'inline--size-' . $value[0]
  ]
}
```